formulae/viewer/routes.py

```diff
@@ -0,0 +1,205 @@
+            else:
+                link_picture=str(images[view])
+
+            return current_app.config['nemo_app'].render(template="viewer::newtabviewer.html", picture=link_picture, objectId=objectId,
+                                                         current_view=view, total_views=len(images), codex=codex, folios=folios, url=dict())
+
+    else:
+        flash( _('This corpus is on copyright, please choose another text'))
+        return current_app.config['nemo_app'].render(template='main::index.html', url=dict())
+
+
+@bp.route("embedded/<objectId>/<view>", methods=["GET"])
+def addviewer(objectId, view):
+
+    if current_app.config['nemo_app'].check_project_team() is True or objectId in current_app.config['nemo_app'].open_texts:
+
+        #information need for loris or
+        collection = current_app.config['nemo_app'].get_collection(objectId)
+        if isinstance(collection, CtsWorkMetadata):
+            editions = [t for t in collection.children.values() if isinstance(t, CtsEditionMetadata)]
+            if len(editions) == 0:
+                raise UnknownCollection('{}'.format(collection.get_label()) + _l(' hat keine Edition.'))
+            objectId = str(editions[0].id)
+        formulae = current_app.picture_file[objectId]
+        passage_data = get_passage(objectId, '1')
+
+
+        #this viewer work when the library or archiv give an IIIF API for the external usage of theirs books
+        if "manifest" in formulae:
+            if "codex" in formulae:
+                codex=formulae["codex"]
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
+                return current_app.config['nemo_app'].render(template='viewer::multiviewermirador.html', manifest=manifest
+                                                             ,objectId=objectId, codex=codex, passage_data=passage_data, url=dict())
+            else:
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
+                with open((current_app.IIIFmapping+"/"+formulae["manifest"]), "r") as f:
+                    title = load(f)
+                codex=title["label"]
+                return current_app.config['nemo_app'].render(template='viewer::multiviewermirador.html', manifest=manifest
+                                                             ,objectId=objectId, codex=codex, passage_data=passage_data, url=dict())
+        else:
+            try:
+                view = int(view)
+            except:
+                #exception when in the url view contains another character than number
+                view=0
+                images = formulae["images"]
+                folios = formulae["folios"]
+                town = formulae['town']
+                codex = formulae['codex']
+                current_folios = folios[view]
+                flash( _('There is not an images for this formula. Showing the first page.'))
```

Comment:
The original messages should always be in German. I would suggest here:

`flash(_('Es gibt kein Bild ') + view + _(' für diese Formel. Das erste Bild wird gezeigt.').`

Also, since the message that you send depends on the original value of `view`, it should go before the line where you set `view = 0`.